src/apple/macos-system-helper.ts

```diff
@@ -0,0 +1,28 @@
+import { execSync } from 'child_process';
+import { debug } from '../utils/debug';
+
+export class MacOSSystemHelpers {
```

Comment:
@Lms24 this utility class is used to resolve paths in Xcode projects. The commands `xcrun` and `xcode-select` come with the installation of the Xcode Command Line Utilities, therefore for Apple platforms developers they will be available.

How should we handle the usage in unit tests when they run on ubuntu in CI?
Should we create fakes for the CI workflows so that `execSync` is actually run, or should we just mock these functions in the javascript tests?

Reply:
Additional option could be skipping tests that are supposed to run on macOS and create another job with `macos-latest` as the `runs-on`

Reply:
You could use fakes in the unit tests for Ubuntu and write some extra integration tests running on MacOS runners. Whatever we do, we should add some tests running on MacOS to ensure this code works correctly.

Reply:
I was just thinking that it might make sense to run the tests in question just on a macos runner then? We can also exclude the unit tests in the general unit test job and executed the specific ones in another job running on macos.

In regards to mocking/if we don't go down the macos runner route: Feel free to do whatever is necessary. No strong opinions. We usually make use of `vi.mock`/`vi.spyOn` for stuff like this, so it's not unheard of 😆

Reply:
To combine your thoughts I'll do this:

- Create a second GHA job to run test suite on macOS too.
- For Linux I'll use fakes so that the entire test suite passes, no need to manage multiple sets of test suites.
- For macOS I'll run the entire test suite, also to ensure we are cross-platform-compatible
- Instead of creating mock CLIs for Linux, I'll use `vi.mock` / `vi.spyOn` to mock `execSync` or similar.
- I'll evaluate making the macOS system helper class an injectable dependency.

Reply:
@philprime, do you plan to do this in this PR, or is there a way to make this happen in follow-up PRs? I would really appreciate getting this PR merged as it's already huge.

Reply:
I need to do this in this PR because the tests can not pass in CI otherwise.